packages/playwright-core/src/client/elementHandle.ts

```diff
@@ -271,8 +271,9 @@ export async function convertInputFiles(files: string | FilePayload | string[] |
     throw new Error('Cannot set buffer larger than 50Mb, please write it to a file and pass its path instead.');
 
   const stats = await Promise.all(items.filter(isString).map(item => fs.promises.stat(item as string)));
+  const totalExceedsMaxSize = stats.reduce((acc, stat) => acc + stat.size, 0) > sizeLimit;
```

Comment:
note: if the size is too big, over multiple files, we fail to convert it to JSON (when it gets stringified, and are not able to send it to the Chromium CDP transport)